README.md

```diff
@@ -1 +1,638 @@
+
+### 1.4.1 Asymmetric Keys
+
+UCAN MUST be used as the signature envelope for AWAKE. Any UCAN-compatible asymmetric key MAY be used for signatures, including RSA, Ed25519, P-256, and so on.
+
+The [ECDH portion](#1431-ecdh-input) of the [Double Ratchet KDF](#143-key-derivation) MUST use P-256.
+
+### 1.4.2 Symmetric Keys
+
+All symmetric encryption in AWAKE MUST use [256-bit AES-GCM](https://csrc.nist.gov/publications/detail/sp/800-38d/final). These keys MUST be generated by the [KDF](#143-key-derivation), and SHOULD be non-extractable where possible.
+
+Each encrypted payload MUST include a unique (freshly generated) 12-byte [initialization vector](https://en.wikipedia.org/wiki/Initialization_vector).
+
+### 1.4.3 Key Derivation
+
+AWAKE uses [HKDF](https://datatracker.ietf.org/doc/html/rfc5869) to derive keys. Key derivation in AWAKE's double ratchet MUST use the following algorithm:
+
+```
+         Diffie-Hellman          Secret             Message
+            Ratchet               Chain             Crypto
+┌──────────────┴─────────────┐  ┌───┴───┐         ┌────┴────┐
+ Alice's P-256    Bob's P-256    Current
+   Private Key    Public Key     Secret
+            │      │                │
+            │      │                │
+      ┌─────┼──────┼────────────────┼────────────────┐
+      │     │      │                │                │
+      │     │      │                │                │
+      │     ▼      ▼                ▼                │
+      │    ┌────────┐          ┌────────┐            │
+      │    │        │          │        │            │
+      │    │  ECDH  ├─────────►│  HKDF  │            │
+      │    │        │          │        │            │
+      │    └────────┘          └────┬───┘            │
+      │                             │                │
+      │                             │                │
+      │                             ▼                │
+      │                        ┌─────────┐  512-603  │
+      │                        │         ├───────────┼───► Unique IV
+      │                        │  Split  │           │
+      │                        │         ├───────────┼───► AES-Key
+      │                        └────┬────┘  256-511  │      (OKM)
+      │                             │                │
+      │                             │ 0-255          │
+      │                             │                │
+      └─────────────────────────────┼────────────────┘
+                                    │
+                                    │
+                                    ▼
+                                  Next
+                                 Secret
+```
+
+``` javascript
+// JS-flavored Pseudocode
+
+const ecdhSecret = ecdh(aliceSk, bobPk)
+const awakeTag = 0x4157414B452D5543414E // "AWAKE-UCAN" as hex
+const pseudorandomBits = hkdf.generateBits({ecdh, salt: currentSecret, info: awakeTag, bitLength: 256 + 256 + 12})
```

Comment:
Should this be `256 + 256 + 96`?

Reply:
How different are bits and bytes _really?_

Reply:
7 bits

Reply:
You're not wrong